core/templates/dev/head/domain/collection/CollectionRightsObjectFactory.js

```
@@ -0,0 +1,78 @@
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Factory for creating and mutating instances of frontend
+ * collection rights domain objects.
+ */
+
+oppia.factory('CollectionRightsObjectFactory', [
+    function() {
+    var CollectionRights = function(collectionRightsObject) {
+      this._collectionId = collectionRightsObject.collection_id;
+      this._canEdit = collectionRightsObject.can_edit;
+      this._canUnpublish = collectionRightsObject.can_unpublish;
+      this._isPrivate = collectionRightsObject.is_private;
+      this._ownerNames = collectionRightsObject.owner_names;
+    };
+
+    // Instance methods
+
+    CollectionRights.prototype.getCollectionId = function() {
+      return this._collectionId;
+    };
+
+    // Returns true if the the user can edit the collection. This property is
+    // immutable.
+    CollectionRights.prototype.canEdit = function() {
+      return this._canEdit;
+    };
+
+    // Returns true if the user can edit the collection. This property is
+    // immutable.
+    CollectionRights.prototype.canUnpublish = function() {
+      return this._canUnpublish;
+    };
+
+    // Returns true if the collection is private.
+    CollectionRights.prototype.isPrivate = function() {
+      return this._isPrivate;
+    };
+
+    CollectionRights.prototype.setPublic = function() {
+      this._isPrivate = false;
+    };
+
+    // Sets isPrivate to true only if canUnpublish in is true.
+    CollectionRights.prototype.setPrivate = function() {
+      if (this.canUnpublish()) {
```

If the user can't unpublish the collection and this method is called, should we be a bit more severe? That seems like an exceptional case to me and worthy of throwing an error or calling into ``alertsService``, rather than silently ignoring it.